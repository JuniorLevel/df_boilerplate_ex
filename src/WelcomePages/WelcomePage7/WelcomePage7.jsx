// @flow

import React from 'react';
import MediaQuery from 'react-responsive';
import { WelcomeButton, SingleImageBg, WelcomeAuthButtons, Welcome } from '@/components';
import { welcomeButtons, authButtons } from './data';
import Image1 from '../pictures/Image1.jpg';

export const WelcomePage7 = (): React.Node => (
	<SingleImageBg imageUrl={Image1}>
		<Welcome buttons={welcomeButtons}>
			<MediaQuery minWidth={361}>
				<WelcomeAuthButtons>
					<WelcomeButton title={authButtons[0].title} onClick={authButtons[0].onClick} />
					<WelcomeButton title={authButtons[1].title} onClick={authButtons[1].onClick} />
					<WelcomeButton title={authButtons[2].title} onClick={authButtons[2].onClick} />
					<WelcomeButton title={authButtons[3].title} onClick={authButtons[3].onClick} />
					<WelcomeButton title={authButtons[4].title} onClick={authButtons[4].onClick} />
					<WelcomeButton title={authButtons[5].title} onClick={authButtons[5].onClick} />
					<WelcomeButton title={authButtons[6].title} onClick={authButtons[6].onClick} />
					<WelcomeButton title={authButtons[7].title} onClick={authButtons[7].onClick} />
					<WelcomeButton title={authButtons[8].title} onClick={authButtons[8].onClick} />
					<WelcomeButton title={authButtons[9].title} onClick={authButtons[9].onClick} />
				</WelcomeAuthButtons>
			</MediaQuery>
			<MediaQuery maxWidth={360}>
				<WelcomeAuthButtons>
					<WelcomeButton title={authButtons[0].title} onClick={authButtons[0].onClick} />
					<WelcomeButton title={authButtons[1].title} onClick={authButtons[1].onClick} />
				</WelcomeAuthButtons>
			</MediaQuery>
		</Welcome>
	</SingleImageBg>
);
