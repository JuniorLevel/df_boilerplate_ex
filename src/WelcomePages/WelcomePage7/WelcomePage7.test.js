import React from 'react';
import { Context as ResponsiveContext } from 'react-responsive';
import { render } from '@testing-library/react';
import { WelcomePage7 } from './WelcomePage7';

describe('WelcomePage7', () => {
	test('should render WelcomePage7 component without crashing', () => {
		const { container } = render(<WelcomePage7 />);
		expect(container).toBeInTheDocument();
	});
	test('matches the snapshot desktop', () => {
		const { container: desktop } = render(
			<ResponsiveContext.Provider value={{ width: 1920 }}>
				<WelcomePage7 />
			</ResponsiveContext.Provider>
		);
		expect(desktop).toMatchSnapshot();
	});
	test('matches the snapshot laptop', () => {
		const { container: laptop } = render(
			<ResponsiveContext.Provider value={{ width: 1024 }}>
				<WelcomePage7 />
			</ResponsiveContext.Provider>
		);
		expect(laptop).toMatchSnapshot();
	});
	test('matches the snapshot mobile', () => {
		const { container: mobile } = render(
			<ResponsiveContext.Provider value={{ width: 360 }}>
				<WelcomePage7 />
			</ResponsiveContext.Provider>
		);
		expect(mobile).toMatchSnapshot();
	});
});
