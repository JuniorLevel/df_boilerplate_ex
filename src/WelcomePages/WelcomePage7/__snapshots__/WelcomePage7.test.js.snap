// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomePage7 matches the snapshot desktop 1`] = `
<div>
  <div
    class="SingleImageBgstyles__DivImageContainer-sc-gojepk-0 YepXU"
  >
    <div
      class="ant-flex css-1c0na6j ant-flex-align-center ant-flex-justify-center"
      style="height: 100vh;"
    >
      <div
        class="ant-layout css-1c0na6j"
        style="width: 100%; max-width: 300px; padding: 20px 15px;"
      >
        <h1
          class="ant-typography css-1c0na6j"
          style="text-align: center;"
        >
          Welcome
        </h1>
        <input
          class="ant-input css-1c0na6j ant-input-outlined"
          placeholder="email"
          style="margin-bottom: 20px;"
          type="text"
          value=""
        />
        <div
          class="Welcome__ButtonContainer-sc-p9wa2l-0 hSXGab"
        >
          <button
            class="ant-btn css-1c0na6j ant-btn-default ant-btn-block"
            type="button"
          >
            <span>
              email
            </span>
          </button>
          <button
            class="ant-btn css-1c0na6j ant-btn-default ant-btn-block"
            type="button"
          >
            <span>
              LogIn
            </span>
          </button>
          <button
            class="ant-btn css-1c0na6j ant-btn-default ant-btn-block"
            type="button"
          >
            <span>
              CreateAccount
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`WelcomePage7 matches the snapshot laptop 1`] = `
<div>
  <div
    class="SingleImageBgstyles__DivImageContainer-sc-gojepk-0 YepXU"
  >
    <div
      class="ant-flex css-1c0na6j ant-flex-align-center ant-flex-justify-center"
      style="height: 100vh;"
    >
      <div
        class="ant-layout css-1c0na6j"
        style="width: 100%; max-width: 300px; padding: 20px 15px;"
      >
        <h1
          class="ant-typography css-1c0na6j"
          style="text-align: center;"
        >
          Welcome
        </h1>
        <input
          class="ant-input css-1c0na6j ant-input-outlined"
          placeholder="email"
          style="margin-bottom: 20px;"
          type="text"
          value=""
        />
        <div
          class="Welcome__ButtonContainer-sc-p9wa2l-0 hSXGab"
        >
          <button
            class="ant-btn css-1c0na6j ant-btn-default ant-btn-block"
            type="button"
          >
            <span>
              email
            </span>
          </button>
          <button
            class="ant-btn css-1c0na6j ant-btn-default ant-btn-block"
            type="button"
          >
            <span>
              LogIn
            </span>
          </button>
          <button
            class="ant-btn css-1c0na6j ant-btn-default ant-btn-block"
            type="button"
          >
            <span>
              CreateAccount
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`WelcomePage7 matches the snapshot mobile 1`] = `
<div>
  <div
    class="SingleImageBgstyles__DivImageContainer-sc-gojepk-0 YepXU"
  >
    <div
      class="ant-flex css-1c0na6j ant-flex-align-center ant-flex-justify-center"
      style="height: 100vh;"
    >
      <div
        class="ant-layout css-1c0na6j"
        style="width: 100%; max-width: 300px; padding: 20px 15px;"
      >
        <h1
          class="ant-typography css-1c0na6j"
          style="text-align: center;"
        >
          Welcome
        </h1>
        <input
          class="ant-input css-1c0na6j ant-input-outlined"
          placeholder="email"
          style="margin-bottom: 20px;"
          type="text"
          value=""
        />
        <div
          class="Welcome__ButtonContainer-sc-p9wa2l-0 hSXGab"
        >
          <button
            class="ant-btn css-1c0na6j ant-btn-default ant-btn-block"
            type="button"
          >
            <span>
              email
            </span>
          </button>
          <button
            class="ant-btn css-1c0na6j ant-btn-default ant-btn-block"
            type="button"
          >
            <span>
              LogIn
            </span>
          </button>
          <button
            class="ant-btn css-1c0na6j ant-btn-default ant-btn-block"
            type="button"
          >
            <span>
              CreateAccount
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
